File: web-development-101/project-euler/largest_prime_factor.js
var isPrime = function(num) {
    for (var i = 2; i < num; i++) {
        if(num%i===0)
            return false;
    }
    return true;
};

var findPrimeFactor = function(num) {
    for(var i = 2; i < num; i++) {
        if(isPrime(i) && num%i === 0)
            return i;
    }
    return 1; // num is prime
};

function largestPrimeFactor(num) {
    var primeFactor = findPrimeFactor(num);

    // base case
    if(primeFactor == 1) {
        return num;
    }
    else {
        return largestPrimeFactor(num/primeFactor);
    }
}

var num = 600851475143;
var ans = largestPrimeFactor(num);

console.log("The largest prime factor of the number " + num + " is " + ans);

File: web-development-101/project-euler/largest_prime_factor.rb
def is_prime num
    for i in 2..num-1 do
        if num%i == 0
            return false
        end
    end
    return true
end

def find_prime_factor num
    for i in 2..num-1 do
        if is_prime(i) and num%i == 0
            return i
        end
    end
    1
end

def largest_prime_factor num
    prime_factor = find_prime_factor(num)

    if prime_factor == 1
        return num
    else
        return largest_prime_factor(num/prime_factor)
    end
end

num = 600851475143
ans = largest_prime_factor num
puts ans